moto/kinesisvideo/models.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import unicode_literals
+from boto3 import Session
+from moto.core import BaseBackend, BaseModel
+from datetime import datetime
+from .exceptions import (
+    ResourceNotFoundException,
+    ResourceInUseException,
+)
+import random
+import string
+
+
+class Stream(BaseModel):
+    def __init__(
+        self,
+        region_name,
+        device_name,
+        stream_name,
+        media_type,
+        kms_key_id,
+        data_retention_in_hours,
+        tags,
+    ):
+        self.region_name = region_name
+        self.stream_name = stream_name
+        self.device_name = device_name
+        self.media_type = media_type
+        self.kms_key_id = kms_key_id
+        self.data_retention_in_hours = data_retention_in_hours
+        self.tags = tags
+        self.status = "ACTIVE"
+        self.version = self._get_random_string()
+        self.creation_time = datetime.utcnow()
+        stream_arn = "arn:aws:kinesisvideo:{}:123456789012:stream/{}/1598784211076".format(
+            self.region_name, self.stream_name
+        )
+        self.data_endpoint_number = self._get_random_hex()
+        self.arn = stream_arn
+
+    def _get_random_hex(self, length=8):
+        chars = list("1234567890abcdef")
+        return "".join([random.choice(chars) for _ in range(length)])
+
+    def _get_random_string(self, length=20):
+        letters = string.ascii_lowercase
+        result_str = "".join([random.choice(letters) for _ in range(length)])
+        return result_str
+
+    def get_data_endpoint(self, api_name):
+        data_endpoint_prefix = "s-" if api_name in ("PUT_MEDIA", "GET_MEDIA") else "b-"
+        return "https://{}{}.kinesisvideo.{}.amazonaws.com".format(
+            data_endpoint_prefix, self.data_endpoint_number, self.region_name
+        )
+
+    def to_dict(self):
+        return {
+            "DeviceName": self.device_name,
+            "StreamName": self.stream_name,
+            "StreamARN": self.arn,
+            "MediaType": self.media_type,
+            "KmsKeyId": self.kms_key_id,
+            "Version": self.version,
+            "Status": self.status,
+            "CreationTime": self.creation_time.isoformat(),
+            "DataRetentionInHours": self.data_retention_in_hours,
+        }
+
+
+class KinesisVideoBackend(BaseBackend):
+    def __init__(self, region_name=None):
+        super(KinesisVideoBackend, self).__init__()
+        self.region_name = region_name
+        self.streams = {}
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+
+    def create_stream(
+        self,
+        device_name,
+        stream_name,
+        media_type,
+        kms_key_id,
+        data_retention_in_hours,
+        tags,
+    ):
+        streams = [_ for _ in self.streams.values() if _.stream_name == stream_name]
+        if len(streams) > 0:
+            raise ResourceInUseException("The stream a already exists.")
```

Comment:
Typo - or should it say `The stream {stream_name} already exists`?

Reply:
Thanks for finding out. updated.
https://github.com/spulec/moto/pull/3271/commits/6384f37b855a6525c041af87fec84eeb12e16e1e